pruner/factory.go

```diff
@@ -0,0 +1,14 @@
+package pruner
+
+import (
+	"context"
+
+	"github.com/celestiaorg/celestia-node/header"
+)
+
+// Factory contains methods necessary to prune data
+// from the node's datastore.
+type Factory interface {
+	AvailabilityWindow
+	Prune(context.Context, ...*header.ExtendedHeader) error
+}
```

Comment:
Could you please provide some details why those needs to be combined? Is it because AvailabilityWindow will have different implementations for LN and FN? II thought they have same recency windowm, so there is no need to make it part of same interface and separate implementation to same thing.

Reply:
The reason why is so we can have an Archival implementation that returns true for every height.

Reply:
I think we can split interfaces, since they are not related to each other:
```
AvailabilityWindow interface - archival / pruned implementations
Prune - light / full implementations
```